hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -135,7 +137,12 @@ private void resolveMutableBuilderAttributes(
             @NonNull final Topic.Builder builder,
             @NonNull final Topic topic) {
         if (op.hasAdminKey()) {
-            builder.adminKey(op.adminKey());
+            var key = op.adminKey();
+            if (isEmpty(key)) {
```

Comment:
Hmm...we would like to _only_ make the topic immutable when the `op.adminKey()` is exactly an empty `KeyList`. (c.f. protobuf message comment [here](https://github.com/hashgraph/hedera-protobufs/blob/main/services/consensus_update_topic.proto#L58)).

On the other hand, it seems possible to use **any** empty `Key` to remove the admin key in `mono-service`; though the EET [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/consensus/TopicUpdateSuite.java#L173) does only test exactly the empty `KeyList`.

I'd suggest adding another EET like the one above that characterizes the current behavior when the empty key is for example, `ThresholdKey{2/2, keys=[KeyList{keys=[]}, KeyList{keys=[]}]}` and seeing what happens. 



I think it would make sense to add an `AttributeValidator.

Reply:
Thanks. I added two more unit tests to test EMPTY_KEYLIST and EMPTY_THRESHOLD_KEY(your mentioned above) in ConsensusUpdateTopicHandlerTest. I also added one test in TopicUpdateSuite to test the ThresholdKey you mentioned. Interestingly, in mono code, EMPTY_THRESHOLD_KEY can remove admin key; in module code, EMPTY_THRESHOLD_KEY can not remove admin key because it's not consider to be KeyUtils.isEmpty(). Is that a correct behavior?

Reply:
Followed the discussion from today's modularization call, I updated the code to clear adminKey with only empty KeyList and their unit tests as well. ThresholdKey{2/2, keys=[KeyList{keys=[]}, KeyList{keys=[]}]} can't clear adminKey in module code. In mono code, if passing this Threholdkey in the transaction, it failed because "No sig control for it", so you can't use it to clear adminKey either.

Reply:
I added AttributeValidator.isImmutableKey(Key key). @tinker-michaelj, Please check if this conversation can be resolved. Thanks.

Reply:
I created a new issue#6690 following the #ask-an-expert question/answer regarding to delete adminKey/autoRenewAccount.